public/js/utils.jsx

```diff
@@ -101,3 +101,60 @@ export function getPayMethodFromUri(payMethods, payMethodUri) {
   }
   return matches.length ? matches[0] : null;
 }
+
+
+export function isValidEmail(email) {
+  if (typeof email !== 'string') {
+    return false;
+  }
+
+  var parts = email.split('@');
+  var localLength = parts[0] && parts[0].length;
+  var domainLength = parts[1] && parts[1].length;
+
+  // Original regexp from:
+  //  http://blog.gerv.net/2011/05/html5_email_address_regexp/
+  // Modified to remove the length checks, which are done later.
+  // IETF spec: http://tools.ietf.org/html/rfc5321#section-4.5.3.1.1
+  // NOTE: this does *NOT* allow internationalized domain names.
+  /* eslint-disable max-len */
```

Comment:
a way to get around linting is to do: `new RegExp('...' + '...' + '...')` which might actually be nice since it would split up the regex into smaller readable chunks.